examples/LIEKF_NavstateExample.cpp

```diff
@@ -0,0 +1,123 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file LIEKF_NavstateExample.cpp
+ * @brief A left invariant Extended Kalman Filter example using the GeneralLIEKF
+ * on NavState using IMU/GPS measurements.
+ *
+ * This example uses the templated GeneralLIEKF class to estimate the state of
+ * an object using IMU/GPS measurements. The prediction stage of the LIEKF uses
+ * a generic dynamics function to predict the state. This simulates a navigation
+ * state of (pose, velocity, position)
+ *
+ * @date Apr 25, 2025
+ * @author Scott Baker
+ * @author Matt Kielo
+ * @author Frank Dellaert
+ */
+#include <gtsam/navigation/NavState.h>
+#include <gtsam/nonlinear/LIEKF.h>
+
+#include <iostream>
+
+using namespace std;
+using namespace gtsam;
+
+// Define a dynamics function.
+// The dynamics function for NavState returns a result vector of
+// size 9 of [angular_velocity, 0, 0, 0, linear_acceleration] as well as
+// a Jacobian of the dynamics function with respect to the state X.
+// Since this is a left invariant EKF, the error dynamics do not rely on the
+// state
+Vector9 dynamics(const NavState& X, const Vector6& imu,
+                 OptionalJacobian<9, 9> H = {}) {
+  const auto a = imu.head<3>();
+  const auto w = imu.tail<3>();
+  Vector9 result;
+  result << w, Z_3x1, a;
+  if (H) {
+    *H = Matrix::Zero(9, 9);

```

Comment:
H->setZero() will be faster.
These dynamics don't depend on the state so maybe we have to rethink this. No sense in multiplying with a bunch of zeros. Let's talk in the meeting.